src/ucp/rndv/rndv_put.c

```diff
@@ -416,21 +425,25 @@ ucp_proto_rndv_put_zcopy_query(const ucp_proto_query_params_t *params,
                       UCP_PROTO_ZCOPY_DESC, put_desc);
 }
 
-static ucs_status_t ucp_proto_rndv_put_zcopy_reset(ucp_request_t *request)
+static ucs_status_t ucp_proto_rndv_put_zcopy_reset(ucp_request_t *req)
 {
-    const ucp_proto_rndv_put_priv_t *rpriv = request->send.proto_config->priv;
-    ucp_datatype_iter_t *dt_iter           = &request->send.state.dt_iter;
-    ucp_lane_map_t atp_sent_map;
+    const ucp_proto_rndv_put_priv_t *rpriv = req->send.proto_config->priv;
 
-    atp_sent_map = rpriv->atp_map & UCS_MASK(request->send.rndv.put.atp_lane);
-    if (atp_sent_map == rpriv->atp_map) {
-        dt_iter->offset = dt_iter->length;
+    if (req->send.rndv.put.atp_count == rpriv->atp_num_lanes) {
```

Comment:
AFAIU we can return `UCS_OK` from `ucp_proto_rndv_put_common_atp_send` if `req->send.rndv.put.atp_count >= req->send.state.dt_iter.length` so it would be considered here as not all ATPs were sent which is wrong.

Reply:
it would mean we sent all ATPs we needed to send, since we should not send more ATPs than the request length